### PR TITLE
Add Animal Pose Preprocessor (RTMPose AP-10K)

### PR DESCRIPTION
**Pull request by sdbds**
#2258 
image

I will training a controlnet-lllite for animal pose, too

### CODE REVIEW

**Comment by sdbds** [comment]
image
image

i tested openpose and dwpose, it works, too.

@huchenlei 
Use animalpose, openpose editer not support it now.

File: annotator/openpose/__init__.py [hunk 1]
```diff
@@ -354,3 +356,36 @@ def __call__(
         if json_pose_callback:
             json_pose_callback(encode_poses_as_json(poses, H, W))
         return draw_poses(poses, H, W, draw_body=include_body, draw_hand=include_hand, draw_face=include_face)
+        
+class AnimalposeDetector:
```
Comment (huchenlei) [inline annotator/openpose/__init__.py:360]: This class is not reference anywhere. How does it interact with rest of the code?

File: annotator/openpose/animalpose.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+cached_onnx_det_name, cached_onnx_pose_name = '', ''
+
+class AnimalPoseImage:
+    def __init__(self, onnx_det: str, onnx_pose: str):
+        global ort_session_det, ort_session_pose, cached_onnx_det_name, cached_onnx_pose_name
+        if check_ort_gpu():
+            import onnxruntime as ort
+            pose_filename = os.path.basename(onnx_pose)
+            if pose_filename != cached_onnx_pose_name and ort_session_pose is None:
+                print(f"AnimalPose: Caching pose session {pose_filename}...")
```
Comment (huchenlei) [inline annotator/openpose/animalpose.py:154]: Use `logger.debug`/`logger.info` instead for pretty printing.

**Comment by huchenlei** [comment]
Hi @sdbds, thanks for the PR! I will give it a test when I get home tonight.

**Comment by sdbds** [comment]
> Hi @sdbds, thanks for the PR! I will give it a test when I get home tonight.
Good night~
I found i chosen wrong branch...i will close this and open new one

**Pull request closed without merging**